patches/export_pdf.patch

```diff
@@ -0,0 +1,131 @@
+diff --git a/pdf/pdf.cc b/pdf/pdf.cc
+index 7aba412..246cd81 100644
+--- a/pdf/pdf.cc
++++ b/pdf/pdf.cc
+@@ -81,21 +81,26 @@ const void* PPP_GetInterface(const char* interface_name) {
+   return pp::Module::Get()->GetPluginInterface(interface_name);
+ }
+ 
++}  // namespace chrome_pdf
++
++
++extern "C" {
++
+ #if defined(OS_WIN)
+-bool RenderPDFPageToDC(const void* pdf_buffer,
+-                                 int buffer_size,
+-                                 int page_number,
+-                                 HDC dc,
+-                                 int dpi,
+-                                 int bounds_origin_x,
+-                                 int bounds_origin_y,
+-                                 int bounds_width,
+-                                 int bounds_height,
+-                                 bool fit_to_bounds,
+-                                 bool stretch_to_bounds,
+-                                 bool keep_aspect_ratio,
+-                                 bool center_in_bounds,
+-                                 bool autorotate) {
++PDF_EXPORT bool RenderPDFPageToDC(const void* pdf_buffer,
++                                  int buffer_size,
++                                  int page_number,
++                                  HDC dc,
++                                  int dpi,
++                                  int bounds_origin_x,
++                                  int bounds_origin_y,
++                                  int bounds_width,
++                                  int bounds_height,
++                                  bool fit_to_bounds,
++                                  bool stretch_to_bounds,
++                                  bool keep_aspect_ratio,
++                                  bool center_in_bounds,
++                                  bool autorotate) {
+   if (!g_sdk_initialized_via_pepper) {
+     if (!chrome_pdf::InitializeSDK()) {
+       return false;
+@@ -118,9 +123,9 @@ bool RenderPDFPageToDC(const void* pdf_buffer,
+ 
+ #endif  // OS_WIN
+ 
+-bool GetPDFDocInfo(const void* pdf_buffer,
+-                   int buffer_size, int* page_count,
+-                   double* max_page_width) {
++PDF_EXPORT bool GetPDFDocInfo(const void* pdf_buffer,
++                              int buffer_size, int* page_count,
++                              double* max_page_width) {
+   if (!g_sdk_initialized_via_pepper) {
+     if (!chrome_pdf::InitializeSDK())
+       return false;
+@@ -176,4 +181,4 @@ bool RenderPDFPageToBitmap(const void* pdf_buffer,
+   return ret;
+ }
+ 
+-}  // namespace chrome_pdf
++}
+diff --git a/pdf/pdf.gyp b/pdf/pdf.gyp
+index f47e33a..6e0a12d 100644
+--- a/pdf/pdf.gyp
++++ b/pdf/pdf.gyp
+@@ -6,7 +6,7 @@
+   'targets': [
+     {
+       'target_name': 'pdf',
+-      'type': 'static_library',
++      'type': 'loadable_module',
```

Comment:
We can change the `type` and `defines` in `chromiumcontent.gypi`, it is preferred not to introduce a new patch if we don't have to, because maintaining patches is very painful when doing upgrade.